servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1499,6 +1540,42 @@ public final Completable flatMapCompletableDelayError(Function<? super T, ? exte
      * @see <a href="http://reactivex.io/documentation/operators/do.html">ReactiveX do operator.</a>
      */
     public final Publisher<T> beforeFinally(Runnable doFinally) {
+        return beforeFinally(TerminalSignalConsumer.from(doFinally));
+    }
+
+    /**
+     * Invokes an appropriate terminal signal of the {@link TerminalSignalConsumer} <strong>before</strong> any of the
+     * following terminal methods are called:
+     * <ul>
+     *     <li>{@link Subscriber#onComplete()} - invokes {@link TerminalSignalConsumer#onComplete()}</li>
+     *     <li>{@link Subscriber#onError(Throwable)} - invokes {@link TerminalSignalConsumer#onError(Throwable)}</li>
+     *     <li>{@link Subscription#cancel()} - invokes {@link TerminalSignalConsumer#onCancel()}</li>
```

Comment:
Here we have a small naming inconsistency between `TerminalSignalConsumer` API and other `when*` operators:
- `whenOnComplete(Runnable)` matches with `TerminalSignalConsumer#onComplete()`;
- `whenOnError(Consumer<Throwable>)` matches with `TerminalSignalConsumer#onError(Throwable)`;

`whenCancel(Runnable)` does not have `On` in the middle to match with `TerminalSignalConsumer#onCancel()`. 

It's ok for me, just want to highlight this naming inconsistency. WDYT?

Reply:
The operator names are following the pattern `before/when/after[actual method name]` so I think `whenCancel()` is correct. We can rename `TerminalSignalConsumer#onCancel()` to `cancel()` to match the pattern

Reply:
`cancel()` sounds like it "cancels" something which seems to be no the case or I am missing something ?

Reply:
Ya its just a conflict between following names of the callbacks these methods represent (`TerminalSignalConsumer#cancel()` here is invoked when `Subscription#cancel()` is called) or changing names to make it more meaningful in this context. We went with following callback names which seems easier to correlate.

Reply:
Renamed to `TerminalSignalConsumer#cancel()`